eam-light-backendweb/src/main/java/ch/cern/cmms/eamlightweb/equipment/EquipmentRest.java

```diff
@@ -59,6 +63,18 @@ public Response readEquipment(@QueryParam("c") String equipment) {
 	public Response createEquipment(Equipment equipment) {
 		Equipment response = null;
 		try {
+            // Generate new numeric code if the requested code starts with @
+            if (equipment.getCode()!=null && equipment.getCode().startsWith("@")) {
+                String prefix = equipment.getCode().substring(1,equipment.getCode().length());
+                Optional<String> newCode = codeGeneratorService.getNextAvailableCode(prefix,
+                    authenticationTools.getInforContext(), "equipmentno", "OSOBJ"+equipment.getTypeCode());
+                if (newCode.isPresent()) {
+                    equipment.setCode(newCode.get());
+                } else {
+                    return badRequest(new Exception("Wrong code provided after '@'"));
+                }
+            }
+            // Create equipment
```

Comment:
This is mostly repeated from PartController. I suggest moving both to CodeGeneratorService, and requiring only the equipment code and what it is (part, asset, position, system), and returning an Optional (the new code or nothing/null).

Reply:
I think we do need to check for the '@' on the controller side, to be able to return the correct Exception?